x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/license/DeleteLicenseRequest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.protocol.xpack.license;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+
+public class DeleteLicenseRequest extends AcknowledgedRequest<DeleteLicenseRequest> {
+
+    public DeleteLicenseRequest() {
```

Comment:
Do we need this, and the `readFrom`/`writeTo` methods? I know this was copy/paste but if we dont need em i vote for cleanup.

Reply:
yes, it's still needed for the transport client

Reply:
I meant did we need the methods override's if all they do is call super :) Not asking if we can remove this from being a `TransportRequest` altogether ;)

I also think that @dakrone did some work while i was on vacation consolidating these to all be just an `AcknowledgedRequest`, instead of actual subclasses that generally do zero extra things. You may ping him to see what he wants to see here, too.

Reply:
Sorry, IIRC the `AcknowledgedResponse` is what @dakrone was reviewing. But I think the constructor, read/write methods can all be deleted and just have a null validate?

Reply:
fixed in fa311cd